File: medplum-java/medplum-java-model/src/main/java/com/fhir/EvidenceVariable_Characteristic.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableEvidenceVariable_Characteristic.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface EvidenceVariable_Characteristic {
  Optional<Reference> definitionReference();

  Optional<Expression> definitionExpression();

  Optional<Period> participantEffectivePeriod();

  Optional<List<Extension>> extension();

  Optional<Duration> participantEffectiveDuration();

  Optional<Timing> participantEffectiveTiming();

  Optional<String> id();

  Optional<Duration> timeFromStart();

  Optional<List<Extension>> modifierExtension();

  Optional<String> definitionCanonical();

  Optional<CodeableConcept> definitionCodeableConcept();

  Optional<String> description();

  Optional<Boolean> exclude();

  Optional<DataRequirement> definitionDataRequirement();

  Optional<List<UsageContext>> usageContext();

  Optional<String> participantEffectiveDateTime();

  Optional<Evidencevariable_characteristicGroupmeasure> groupMeasure();

  Optional<TriggerDefinition> definitionTriggerDefinition();

  static ImmutableEvidenceVariable_Characteristic.Builder builder() {
    return ImmutableEvidenceVariable_Characteristic.builder();
  }
}

File: medplum-typescript/src/model/fhir/EvidenceVariable_Characteristic.ts
import { CodeableConcept } from "./CodeableConcept";
import { DataRequirement } from "./DataRequirement";
import { Duration } from "./Duration";
import { Evidencevariable_characteristicGroupmeasure } from "./Evidencevariable_characteristicGroupmeasure";
import { Expression } from "./Expression";
import { Extension } from "./Extension";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { Timing } from "./Timing";
import { TriggerDefinition } from "./TriggerDefinition";
import { UsageContext } from "./UsageContext";

export interface EvidenceVariable_Characteristic {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    description: string | null | undefined;
    definitionReference: Reference | null | undefined;
    definitionCanonical: string | null | undefined;
    definitionCodeableConcept: CodeableConcept | null | undefined;
    definitionExpression: Expression | null | undefined;
    definitionDataRequirement: DataRequirement | null | undefined;
    definitionTriggerDefinition: TriggerDefinition | null | undefined;
    usageContext: UsageContext[] | null | undefined;
    exclude: boolean | null | undefined;
    participantEffectiveDateTime: string | null | undefined;
    participantEffectivePeriod: Period | null | undefined;
    participantEffectiveDuration: Duration | null | undefined;
    participantEffectiveTiming: Timing | null | undefined;
    timeFromStart: Duration | null | undefined;
    groupMeasure:
        | Evidencevariable_characteristicGroupmeasure
        | null
        | undefined;
}
